### ISSUE TITLE
Is it planned to offer other backends than dropbox?

### ISSUE THREAD

**Issue by serapath** [robertknight/passcards#52]
Maybe something, that I can be self hosted?

**Comment by robertknight** [comment]
Yes. For the initial beta release I'm intending to only promote the Dropbox backend because it is well tested but Passcards is designed to allow easy addition of other storage services. The command-line version already uses this facility to work with the local file system instead of Dropbox for example.

As for self hosting, I'm currently doing some work on a simple [HTTP server backend](https://github.com/robertknight/passcards/blob/master/lib/vfs/http.ts) which can be run locally. This is mainly for automated testing and offline development but with the addition of SSL support and a login page it could be turned into a feature for advanced users to be able to self-host their storage.

Implementing a new backend is done by implementing the [VFS](https://github.com/robertknight/passcards/blob/master/lib/vfs/vfs.ts#L68) API, which is a simple file system access abstraction.

**Comment by robertknight** [comment]
I've just landed much of the work to make this possible. Initially this enables multiple stores associated with different Dropbox accounts, which you can use for basic password sharing for example.

For a self-hosted option what I'd ideally like is an open source file server that exposes a REST API and supports OAuth authentication. Open to other suggestions though.

**Comment by serapath** [comment]
Not sure if this fullfills the needs, but it might be a good starting point
https://www.npmjs.com/package/accountdown
